tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -1640,6 +1640,13 @@ uint64_t FragmentMetadata::tile_size(
                       cell_num * array_schema_->cell_size(name);
 }
 
+uint64_t FragmentMetadata::tile_validity_size(
+    const std::string& name, uint64_t tile_idx) const {
+  auto is_nullable = array_schema_->is_nullable(name);
+  auto cell_num = this->cell_num(tile_idx);
+  return (is_nullable) ? cell_num * sizeof(uint8_t) : 0;
+}
```

Comment:
This function is called in a loop where the value of `name` is constant. `ArraySchema::is_nullable` is doing a hash table lookup each time this function is called, and it always returns the same answer. That's an unnecessary inefficiency.

Implementing `FragmentMetadata::max_tile_size` would be better code structure overall, and make it easier to eliminate the inefficiency without insisting that the structure of fragment metadata be exposed to `class Array`.